src/Mammon/Actors/LAWorkspaceActor.cs

```diff
@@ -1,54 +1,66 @@
-﻿namespace Mammon.Actors;
+﻿using Azure.ResourceManager.Resources.Models;
+
+namespace Mammon.Actors;
 
 public class LAWorkspaceActor(ActorHost actorHost, ILogger<LAWorkspaceActor> logger, CostCentreService costCentreService, CostCentreRuleEngine costCentreRuleEngine) : ActorBase<CoreResourceActorState>(actorHost), ILAWorkspaceActor
 {
 	private static readonly string CostStateName = "laWorkspaceCostState";
 
-	public async Task SplitCost(string reportId, string resourceId, ResourceCost totalCost, IEnumerable<LAWorkspaceQueryResponseItem> data)
+	public async Task SplitCost(SplittableResourceRequest request, IEnumerable<LAWorkspaceQueryResponseItem> data)
 	{
-		ArgumentException.ThrowIfNullOrWhiteSpace(resourceId);
-		ArgumentException.ThrowIfNullOrWhiteSpace(reportId);
-		ArgumentNullException.ThrowIfNull(data);
-		ArgumentNullException.ThrowIfNull(totalCost);
+		ArgumentNullException.ThrowIfNull(request);
+
+		var totalCost = request.Resource.Cost;
+		var resourceId = request.Resource.ResourceId;
+		var tags = request.Resource.Tags;
+		var reportId = request.ReportRequest.ReportId;
```

Comment:
I think you could do something like 
`var (totalCost, resourceId, reportId) = request.ReportRequest;` or something.. I might be wrong.

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct

Reply:
I don't think this applies here.. request and its members are not tuples

I do tuple deconstruction at other structures where I actually return tuple